src/shogun/statistical_testing/internals/mmd/ComputeMMD.h

```diff
@@ -108,13 +108,14 @@ struct ComputeMMD
 		const BlockXt& b_xy=map.block(m_n_x, 0, m_n_y, m_n_x);
 
 		terms_t terms;
-		terms.diag[0]=b_x.diagonal().sum();
-		terms.diag[1]=b_y.diagonal().sum();
-		terms.diag[2]=b_xy.diagonal().sum();
+		// cast to 64 bit to avoid overflows
```

Comment:
kernel matrices are often stored as `float32_t`, so the `.sum()` calls over the whole matrix here overflow